knox/views.py

```diff
@@ -39,19 +39,26 @@ def post(self, request, format=None):
                     status=status.HTTP_403_FORBIDDEN
                 )
         token_ttl = self.get_token_ttl()
-        token = AuthToken.objects.create(request.user, token_ttl)
+        instance, token = AuthToken.objects.create(request.user, token_ttl)
         user_logged_in.send(sender=request.user.__class__,
                             request=request, user=request.user)
         UserSerializer = self.get_user_serializer_class()
+
         if UserSerializer is None:
-            return Response({
+            data = {
+                'token': token
+            }
+        else:
+            context = self.get_context()
+            data = {
+                'user': UserSerializer(
+                    request.user,
+                    context=context
+                ).data,
                 'token': token
-            })
-        context = self.get_context()
-        return Response({
-            'user': UserSerializer(request.user, context=context).data,
-            'token': token,
-        })
+            }
+        data['expiry'] = instance.expires
```

Comment:
My thinking was keep it dry here

Reply:
works for me and I like the `expiry` name 🤓

Reply:
could we turn it around to save some lines?

like this (no need to copy it literally):

```
data = {"expiry": instance.expires, "token": token}
if UserSerializer is not None:
   data["user"] = ...
```

what i personally dont like is that we then have 2 names for the same thing, which is always confusing and frustrating for someone learning: `expiry` on the response vs. `expires` on the model.
I also like `expiry` better for the reasons someone else has mentioned. and we could in the same instant change the model field name too as we make this a major release already anyway.

what do you think? :)

Reply:
I'm all in for the field rename, I don't like it.
Putting a DB migration on the token expiry should again be properly advertised in the changelog, I'm not sure how alter table for a rename works in other DB engine but I know postgres would take a full lock on a rename operation...
You definitely would want to take appropriate action to make sur that it's at least as fast as possible :-)

Reply:
Hello guys,
I fully agree with you, we should change the fieldname from `expires` to `expiry`, its more clean. Latest commit have these changes, however @johnraz could you elaborate what you meant with "making sure its as fast as possible"? Afaik this is only a field rename, ought to be pretty fast no?

Relevant docs: 
- https://docs.djangoproject.com/en/2.1/ref/migration-operations/#renamefield
- https://docs.djangoproject.com/en/2.1/_modules/django/db/migrations/operations/fields/#RenameField

Reply:
@sphrak in most scenarios it should be fast enough indeed, with a very large token table though it could lock the table for a fairly annoying amount of time but I'm probably overthinking this.

I was thinking about investigating a solution like copying the data to the new column name and then dropping the old one instead of renaming, to avoid the lock. But I have not thought about this much longer and can't say if it would bring any improvements over the rename, I'm glad this is now merged btw 🎉

Reply:
@johnraz I'm not entirely sure but I would assume in the case of postgres that such a "migration" like renaming a column/field is translated directly into sql syntax `RENAME COLUMN column_name TO new_column_name;` which ought to be much faster in postgres C code than running a data migration in python -- more so at least under the conditions you where concerned about.

but yes, time will tell if we made a wopsy but I am also happy its merged now. Again thank you both for the code reviews, I really enjoy working with you guys. :grinning: 

On a different note (working on the refresh endpoint PR), you dont happen to know if I can reload the `urls.py` via `reload_module` in testing? I cannot get it to work. I am trying to implement a conditional url setting so that we only expose the refresh endpoint if the user has `AUTO_REFRESH` and `TOKEN_TTL` set to not `None` otherwise it wont make sense to have it exposed.

We can continue the discussion in #156 :)

Reply:
I didn't mean a python data migration as it's obviously slower, you can run arbitrary SQL queries in Django migration. I was more thinking on how to compose those SQL queries to achieve a non blocking solution.

I'm not sure about `reload_module` I haven't been touching the tests for a while maybe @belugame knows better but yea let's keep discussing on the other ticket 😉